bitwarden_license/src/Commercial.Core/SecretsManager/Commands/Secrets/UpdateSecretCommand.cs

```diff
@@ -28,6 +28,11 @@ public async Task<Secret> UpdateAsync(Secret updatedSecret, Guid userId)
             throw new NotFoundException();
         }
 
+        if (updatedSecret.Projects != null && updatedSecret.Projects.Any() && !(await _projectRepository.ProjectsAreInOrganization(updatedSecret.Projects.Select(p => p.Id).ToList(), secret.OrganizationId)))
+        {
+            throw new NotFoundException();
+        }
+
```

Comment:
Same thing here, either add a check that the project list only contains one project, or alter the access check to check all the projects provided.

Reply:
Same as CreateSecretCommand comment!